src/archive.cpp

```diff
@@ -85,14 +85,19 @@ namespace zim
 
   entry_index_type Archive::getMediaCount() const
   {
-    return countMimeType(
-      getMetadata("Counter"),
-      [](const std::string& mimetype) {
-        return mimetype.find("image/") == 0 ||
-               mimetype.find("video/") == 0 ||
-               mimetype.find("audio/") == 0;
-      }
-    );
+    try {
+      return countMimeType(
+        getMetadata("Counter"),
+        [](const std::string& mimetype) {
+          return mimetype.find("image/") == 0 ||
+                 mimetype.find("video/") == 0 ||
+                 mimetype.find("audio/") == 0;
+        }
+      );
+    } catch(const EntryNotFound& e) {
+      return (m_impl->getNamespaceEntryCount('I').v
+            + m_impl->getNamespaceEntryCount('J').v);
+    }
```

Comment:
This workaround addresses (in the sense of attempting to return a correct value) only old zimfiles (utilizing the old namespace scheme). For new ZIM-files that for some reason lack the `Counter` metadata entry, the result will be `0`. Is that OK?

Reply:
It is ok to me. `Counter` is necessary to count. If you don't have it, it seems ok to me to return inconsistent/false value. On top of that, libzim is automatically create the `M/Counter` entry so it should be pretty rare.

@kelson42 is it ok for you ?

Reply:
Ok